pkg/controller/policy/policy.go

```diff
@@ -108,7 +108,7 @@ func namespaceEventHandler() handler.Funcs {
 		},
 		// When a namespace is deleted all the resources in the namespace are deleted
 		// but we trigger the reconciliation to eventually perform some cluster-wide cleanup if necessary
-		DeleteFunc: func(ctx context.Context, e event.DeleteEvent, q workqueue.RateLimitingInterface) {
+		DeleteFunc: func(ctx context.Context, e event.TypedDeleteEvent[client.Object], q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
```

Comment:
nit: `event.DeleteEvent` is an alias for `event.TypedDeleteEvent[client.Object]`, we can keep it if we want.